### ISSUE TITLE
How to create extra disk to test-kitchen VM

### ISSUE THREAD

**Issue by krishus** [test-kitchen/kitchen-azurerm#60]
I was wondering does chef test-kitchen for azure support creating extra disk??. If so what step I need to add in .kitchen.yml file??

**Comment by stuartpreston** [comment]
Hi @krishus - no, this is currently not supported. Do you have a requirement to add additional disks?  How would you like to see it represented in the .kitchen.yml?  Would you like 1 disk or more than 1?  Would they be Managed Disks or unmanaged (Storage Account) disks?

**Comment by krishus** [comment]
oh okay. Actually we have a option to add OS disk right by using os_disk_size_gb in .kitchen.yml. In that way if possible I would like to add one or more data disks to my test-kitchen instance whatever it is managed or unmanaged disks.....

**Comment by bcolebank** [comment]
We have the same need to add additional disks via Test Kitchen.
We typically use managed disk for our systems in Azure.
We need the ability to add 1 as well as more than 1 for different machine configuration.

**Comment by stuartpreston** [comment]
I've been working on some of this.  Where I'm at right now is that you can specify a list of data_disks in your `.kitchen.yml` like so:

```
platforms:
- name: windows2016
  driver_config:
    image_urn: MicrosoftWindowsServer:WindowsServer:2016-Datacenter:latest
    data_disks:
      - lun: 0
        disk_size_gb: 64
      - lun: 1
        disk_size_gb: 128
```
On Windows, in addition to configuring WinRM, we will also initialize and format these disks, starting from drive letter F: (drive D: being your temporary disk and drive E: usually being reserved for CDROM).

![image](https://user-images.githubusercontent.com/6384654/29378863-8ef3663e-82b8-11e7-95c1-989be4d7901d.png)

![image](https://user-images.githubusercontent.com/6384654/29378892-a15841b4-82b8-11e7-87fe-fd1744f8f299.png)

This would be applicable for Managed Disks only in the first instance. For Linux, the drives will be attached but uninitialized/formatted/mounted.  As I expect most recipe development on Linux will be fine using the default filesystem layout.

Feel free to add to this thread if you think there's something I'm missing but I hope to get this out soon.

**Comment by krishus** [comment]
Thank you @stuartpreston that would be awesome!

**Comment by abmptit** [comment]
Hi @stuartpreston , since each version adding disk is working on kitchen ?
I'm trying to add serate disk with ubuntu 16 with this conf :
PS > chef -v
Chef Development Kit Version: 2.0.26
chef-client version: 13.2.20
delivery version: master (17c1b0fed9be4c70f69091a6d21a4cbf0df60a23)
berks version: 6.2.0
kitchen version: 1.16.0
inspec version: 1.30.0
Thanks.

**Comment by stuartpreston** [comment]
Version 0.12.0 will be the first version to support the additional disks. Look out for it in the next 1-2 weeks.

**Comment by bcolebank** [comment]
That is great news!

**Comment by anniehedgpeth** [comment]
Hey @stuartpreston , just wanted to check in and see how this is coming along? No rush at all! I just wanted to gauge how I should time my Jira issues related to this. :)

**Comment by stuartpreston** [comment]
This has just been released to Rubygems as version 0.12.0 - I look forward to your feedback!

**Comment by nhudacin** [comment]
You are awesome! Thank you for this!
